<!-- Velocity controller from this PR https://github.com/ethz-asl/rotors_simulator/pull/293/files -->
<launch>
  <arg name="mav_name" default="bebop2"/>
  <arg name="world_name" default="floating_marker"/>
  <!-- <arg name="world_name" default="lar"/> -->
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="log_file" default="$(arg mav_name)" />
  <arg name="dev" default="/dev/input/js2" />
  <arg name="debug" default="false"/>
  <arg name="gui" default="true"/>
  <arg name="paused" default="false"/>
  <!-- The following line causes gzmsg and gzerr messages to be printed to the console
      (even when Gazebo is started through roslaunch) -->
  <arg name="verbose" default="false"/>

  <env name="GAZEBO_MODEL_PATH" value="$(find rotors_gazebo)/models"/>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find rotors_gazebo)/worlds/$(arg world_name).world"/>
    <arg name="debug" value="$(arg debug)" />
    <arg name="paused" value="$(arg paused)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="verbose" value="$(arg verbose)"/>
  </include>

  <group ns="$(arg mav_name)">
    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="$(arg mav_name)" />
      <!-- <arg name="model" value="$(find rotors_description)/urdf/$(arg mav_name)_generic_odometry_sensor.gazebo" /> -->
      <!-- <arg name="model" value="$(find rotors_description)/urdf/mav_with_velodyne_puck_sensor.gazebo" /> -->
      <arg name="model" value="$(find rotors_description)/urdf/mav_generic_odometry_sensor.gazebo" />
      <arg name="enable_logging" value="$(arg enable_logging)" />
      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
      <arg name="log_file" value="$(arg log_file)"/>
      <arg name="x" value="0.0"/>
      <arg name="y" value="0.0"/>
      <arg name="z" value="0.3"/>
      <arg name="yaw" value="0.0"/>
    </include>
    <!-- <node name="joy_node" pkg="joy" type="joy_node">
      <param name="dev" value="$(arg dev)" />
      <param name="dev_name" value="Sony Interactive Entertainment Wireless Controller" />
    </node> -->
    <!-- <node name="ps4_controller_node" pkg="Interface_Bebop" type="Ps4ControllerNode.py" output="screen">
      <remap from="/joy"  to="/bebop2/joy" />
    </node> -->

    <node name="fake_driver" pkg="rotors_gazebo" type="fake_driver" output="screen" >
      
      <!--<remap from="~odom" to="odometry_sensor1/odometry" />-->
      <rosparam command="load" file="$(find rotors_gazebo)/resource/ps3.yaml" />
      <remap from="~odom" to="ground_truth/odometry" />
      <remap from="~command/trajectory" to="command/trajectory" />
      <rosparam> 
        max_vel: 3.0
        max_yawrate: 0.66
      </rosparam>
    </node>
    <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
      <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_$(arg mav_name).yaml" />
      <rosparam command="load" file="$(find rotors_gazebo)/resource/$(arg mav_name).yaml" />
      <remap from="odometry" to="ground_truth/odometry" />
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
    <node name="kalman_filter_node" pkg="interface_bebop" type="Kalman.py" />
  </group>    
    <node name="RobotPoseInWorld" pkg="interface_bebop" type="RobotPoseInWorldtf.py" >      
      <param name="TagPx" value="0.7" />
      <param name="TagPy" value="0.0" />
      <param name="TagPz" value="1.0" />      
      <param name="TagRoll" value="0.0" />  
      <param name="TagPitch" value="-1.57" /> 
      <param name="TagYaw" value="0.0" />  
      <remap from="/robot_aruco_pose" to="/natnet_ros/Bebop1/pose" />
    </node>
    
    <!-- <node name="RobotPoseInWorld" pkg="Interface_Bebop" type="RobotPoseInWorldtf.py" >       
    </node> -->
    <include file="$(find interface_bebop)/launch/monosingle.launch" />
</launch>
